### PR TITLE
Fixes #13634: file_from_shared_folder does not report when trying to copy a directory

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/13634

### CODE REVIEW

File: tree/30_generic_methods/file_copy_from_remote_source_recursion.cf [hunk 1]
```diff
@@ -71,6 +71,12 @@ bundle agent file_copy_from_remote_source_recursion(source, destination, recursi
         classes      => classes_generic_two("${old_class_prefix}", "${class_prefix}");
 
   methods:
+    iteration_2::
```
Comment (ncharles) [inline tree/30_generic_methods/file_copy_from_remote_source_recursion.cf]: can you add a comment on why adding this condition ? I'm pretty sure future us won't remember why the hell we define and error class

**Comment by Fdall** [comment]
OK, merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

File: tree/30_generic_methods/file_copy_from_remote_source_recursion.cf [hunk 2]
```diff
@@ -71,6 +71,16 @@ bundle agent file_copy_from_remote_source_recursion(source, destination, recursi
         classes      => classes_generic_two("${old_class_prefix}", "${class_prefix}");
 
   methods:
+    # A known issue prevents cfengine to raise classes
+    # when dealing with empty promiser.
+    # Applying this bundle on a directory with a 0 level of recursion,
+    # no promisers will be found and we will need to define classes by ourselves.
+    iteration_2::
+      "error" usebundle => _classes_failure("${old_class_prefix}"),
+        ifvarclass=> "!${class_prefix}_reached";
```
Comment (ncharles) [inline tree/30_generic_methods/file_copy_from_remote_source_recursion.cf]: it should have been  ifvarclass=> "!${old_class_prefix}_reached";
class_prefix is null :/

**Pull request merged**